### PR TITLE
fix/compress trivy scan result

### LINKED ISSUE #422: use compression to reduce trivy output report size to avoid hitting container logs max size
It has been found out that trivy scan job produces scanning results bigger than pod max log default value (10MB), as result log rotation happens and operator is unable to unmarshal results from pod logs.

- suggested  workaround : Increase the [kubelet default ](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/) `--container-log-max-size`

- long-term solution will be to write the report output to the pod filesystem and stream it 
Related: #206

### PR DESCRIPTION
**Pull request by chen-keinan**
## Related issues
- Close #422 

- Related #206

## Checklist
- [x] I've read the [guidelines for contributing](https://github.com/aquasecurity/trivy-operator/blob/main/CONTRIBUTING.md) to this repository.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy-operator/tree/main/docs) with the relevant

### CODE REVIEW

File: pkg/plugins/trivy/plugin.go [hunk 1]
```diff
@@ -661,17 +664,10 @@ func (p *plugin) getPodSpecForStandaloneMode(ctx trivyoperator.PluginContext, co
...
 			Args: []string{
-				"--cache-dir",
-				"/tmp/trivy/.cache",
-				"--quiet",
-				"image",
-				"--skip-update",
-				"--format",
-				"json",
-				optionalMirroredImage,
+				"-c", fmt.Sprintf("trivy image %s --cache-dir /tmp/trivy/.cache --quiet  --skip-update --format json > /tmp/scan/result.json &&  bzip2 -c /tmp/scan/result.json | base64", optionalMirroredImage),
```
Comment (JAORMX) [inline pkg/plugins/trivy/plugin.go:670]: Has the value of `optionalMirroredImage` been validated beforehand? Just making sure someone doesn't do anything funky with that now that we're using it for string interpolation.

File: pkg/plugins/trivy/plugin.go [hunk 2]
```diff
@@ -1026,16 +1019,10 @@ func (p *plugin) getPodSpecForClientServerMode(ctx trivyoperator.PluginContext,
...
 			},
 			Args: []string{
-				"--quiet",
-				"image",
-				"--format",
-				"json",
-				"--server",
-				trivyServerURL,
-				optionalMirroredImage,
+				"-c", fmt.Sprintf("trivy image %s --quiet --format json --server %s > /tmp/scan/result.json &&  bzip2 -c /tmp/scan/result.json | base64", optionalMirroredImage, trivyServerURL),
```
Comment (JAORMX) [inline pkg/plugins/trivy/plugin.go:1025]: have the `optionalMirroredImage` and `trivyServerURL` been validated beforehand? Someone could insert arbitrary commands through those.

**Review by JAORMX — CHANGES_REQUESTED** [review]
this LGTM, I just have a couple of minor questions that I left in-line, and that's it.

File: pkg/plugins/trivy/plugin.go (hunk 2 above)
Comment (chen-keinan) [reply pkg/plugins/trivy/plugin.go:1025]: @JAORMX thanks for the input , looking at it.

**Review by JAORMX — APPROVED** [review]
/lgtm

File: pkg/utils/compressutil.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	_, err := io.Copy(uncompressedWriter, bz2Reader)
+	if err != nil {
+		return nil, err
+	}
+	return uncompressedWriter, nil
+}
+
+// Base64Decode accept encoded reader and base64 decode it
+func Base64Decode(EncodedReader io.Reader) ([]byte, error) {
+	EncodedBytes, err := ioutil.ReadAll(EncodedReader)
```
Comment (josedonizetti) [inline pkg/utils/compressutil.go:25]:
```suggestion
	encodedBytes, err := ioutil.ReadAll(EncodedReader)
```

**Review by josedonizetti — APPROVED** [review]
only found a typo

**Review by JAORMX — APPROVED** [review]

**Comment by chen-keinan** [comment]
holding with this PR as we hit the `etcd` limit `etcdserver: request is too large` issue

File: pkg/utils/compressutil.go [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+	uncompressedWriter := new(bytes.Buffer)
+	_, err := io.Copy(uncompressedWriter, bz2Reader)
+	if err != nil {
+		return nil, err
+	}
+	return uncompressedWriter, nil
+}
+
+// Base64Decode accept encoded reader and base64 decode it
+func Base64Decode(EncodedReader io.Reader) ([]byte, error) {
```
Comment (josedonizetti) [inline pkg/utils/compressutil.go:24]: typo

```suggestion
func Base64Decode(encodedReader io.Reader) ([]byte, error) {
```

**Comment by JAORMX** [comment]
LGTM

shipit! 😄

**Pull request merged**